src/main/java/org/folio/marccat/integration/tools/TenantLoading.java

```diff
@@ -154,54 +161,77 @@ public void perform(TenantAttributes ta, Map<String, String> headers) {
-                        Map<String, String> headers, Iterator<LoadingEntry> it) {
+  private void performR(String okapiUrl, Map<String, String> headers, boolean loadBibliographicSample) {
 
-    loadData(okapiUrl, headers);
+    if (loadBibliographicSample) {
+      loadDataBibliographic(okapiUrl, headers);
+    }
+    loadDataAuthority(okapiUrl, headers);
   }
+
   /**
    * Load data.
    *
    * @param okapiUrl the okapi url
-   * @param headers the headers
-    */
-  private void loadData(String okapiUrl, Map <String, String> headers) {
+   * @param headers  the headers
+   */
+  private void loadDataBibliographic(String okapiUrl, Map<String, String> headers) {
     final String endPointUrl = okapiUrl + "/" + "marccat/load-from-file";
     logger.debug("Load data URL " + endPointUrl);
     HttpHeaders httpHeaders = new HttpHeaders();
     httpHeaders.add(Global.OKAPI_TENANT_HEADER_NAME, headers.get(Global.OKAPI_TENANT_HEADER_NAME));
     httpHeaders.add(Global.OKAPI_URL, headers.get(Global.OKAPI_URL));
     httpHeaders.setContentType(MediaType.MULTIPART_FORM_DATA);
-    MultiValueMap <String, Object> requestEntity = new LinkedMultiValueMap <>();
+    MultiValueMap<String, Object> requestEntity = new LinkedMultiValueMap<>();
     File file = getResourceAsFile("/sample-data/load-from-file/sample_records.mrc");
     requestEntity.add("files", new FileSystemResource(file));
-    HttpEntity <MultiValueMap <String, Object>> httpEntity = new HttpEntity <>(requestEntity, httpHeaders);
-    client.exchange(
-      fromUriString(endPointUrl)
-        .build()
-        .toUri(),
-      HttpMethod.POST,
-      httpEntity,
-      String.class);
+    HttpEntity<MultiValueMap<String, Object>> httpEntity = new HttpEntity<>(requestEntity, httpHeaders);
+    client.exchange(fromUriString(endPointUrl).build().toUri(), HttpMethod.POST, httpEntity, String.class);
+
+  }
+
+  /**
+   * Load data.
+   *
+   * @param okapiUrl the okapi url
+   * @param headers  the headers
+   */
+  private void loadDataAuthority(String okapiUrl, Map<String, String> headers) {
+    final String endPointUrl = okapiUrl + "/marccat/authority-record";
+    logger.debug("Load data URL " + endPointUrl);
+    HttpHeaders httpHeaders = new HttpHeaders();
+    httpHeaders.add(Global.OKAPI_TENANT_HEADER_NAME, headers.get(Global.OKAPI_TENANT_HEADER_NAME));
+    httpHeaders.add(Global.OKAPI_URL, headers.get(Global.OKAPI_URL));
+    httpHeaders.setContentType(MediaType.APPLICATION_JSON);
+
+    String requestJson;
+    try {
+      requestJson = IOUtils.toString(this.getClass().getResourceAsStream("/authority/name.json"),
+          String.valueOf(StandardCharsets.UTF_8));
+
+      HttpEntity<String> entity = new HttpEntity<>(requestJson, httpHeaders);
+      client.postForObject(fromUriString(endPointUrl).build().toUri(), entity, String.class);
+    } catch (IOException e) {
+      logger.error(Message.MOD_MARCCAT_00013_IO_FAILURE, e);
```

Comment:
Is this data loading initiated via an API call? If so, how is this error reported back to the client?

Reply:
I have checked if the api returns an error